tests/jballerina-unit-test/src/test/resources/test-src/annotations/deprecation_annotation.bal

```diff
@@ -390,7 +390,73 @@ type Employee5 record {
 };
 
 public function testDeprecatedAnonStructAsStructField() {
-    Employee5 employee5 = {address: {}};
+    Employee5 employee5 = {address: {}};    // warning
     _ = employee5.address;  // warning
     _ = employee5.address.line02;   // warning
 }
+
+type Company record {
+    int companyId;
+
+    @deprecated
+    string name;
+
+    @deprecated
+    string city;
+
+    @deprecated
+    string country;
+};
+
+public function testDeprecatedAnonRecordFieldInInitialization() {
+     Job _ = {
+        title: "SE",
+        experience: 1  // warning
+    };
+
+    int experience = 2;
+    Job _ = {
+        title: "SE",
+        experience  // warning
+    };
+
+    var details = {experience: 2};
+    Job _ = {
+        title: "SE",
+        ...details  // warning
+    };
+
+    string city = "Berlin";
+    var companyDetails = {country: "Germany"};
+    Company _ = {
+        companyId: 1,
+        name: "Foo",    // warning
+        city: city,     // warning
+        ...companyDetails   // warning
+    };
+}
+
+function fooFn(int x1, @deprecated int x2){
+}
+
+function barFn(string s, *Company company) {
+}
+
+public function testDeprecatedParamUsages() {
+    fooFn(10, 11);
+    fooFn(12, x2 = 13);
+    fooFn(x1 = 14, x2 = 15);
+    barFn("bar", {
+        companyId: 1,
+        city: "Berlin", // Warning
+        country: "Germany", // Warning
+        name: "Bar" // Warning
+    });
+    barFn(
+        "bar",
+        companyId = 1,
+        city = "Berlin", // Warning
+        country = "Germany", // Warning
+        name = "Bar" // Warning
```

Comment:
Can we add tests with different ways/combinations for calling the function? E.g.,

- args in rest arg
- named args for required params
- positional args for defaultable params

Reply:
> * args in rest arg

Can you point to the tests added for this please?

Reply:
Here. https://github.com/dulajdilshan/ballerina-lang/blob/0b9d0b06fdad1849da548dac067d8096f22214ef/tests/jballerina-unit-test/src/test/resources/test-src/annotations/deprecation_annotation.bal#L476

Reply:
That's not a rest arg? https://ballerina.io/spec/lang/master/#rest-arg

Reply:
My bad. I thought you mean positional args where we have rest param.